### PR TITLE
Reorganize sidebar nav

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/18143

1. Changes Library -> Components
1. Moves `Usage` documents into `Designer Documentation`
    - This could even be `Usability`
1. Fixes the `AlertBox` designer documentation
    - Turns out it was broken all along...

## Screenshots
image

**Note:** `Form controls` is removed in [another PR](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/566). I'll have to resolve the merge conflicts in one PR or the other when the time comes.

### CODE REVIEW

File: packages/formation-react/stories/address-block.stories.mdx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
 
-<Meta title="Library/Address block" />
+<Meta title="Components/Address block" />
```
Comment (bkjohnson) [inline packages/formation-react/stories/address-block.stories.mdx:3]: Should this go under `Designer Documentation`?

File: packages/formation-react/stories/form-controls.stories.mdx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Canvas, Meta } from '@storybook/addon-docs/blocks';
 
-<Meta title="Library/Form controls/Usage" />
+<Meta title="Components/Form controls/Usage" />
```
Comment (bkjohnson) [inline packages/formation-react/stories/form-controls.stories.mdx:3]: This one as well might belong in `Designer Documentation`.

File: packages/formation-react/src/components/Breadcrumbs/usage.stories.mdx [hunk 3]
```diff
@@ -1,62 +0,0 @@
-import { Meta } from '@storybook/addon-docs/blocks';
-
-<Meta title="Library/Breadcrumbs/Usage" />
```
Comment (bkjohnson) [inline packages/formation-react/src/components/Breadcrumbs/usage.stories.mdx:3]: Why was this deleted?

**Review by bkjohnson — COMMENTED** [review]
Just a few comments about shifting things around.

File: packages/formation-react/stories/address-block.stories.mdx (hunk 1 above)
Comment (cvalarida) [reply packages/formation-react/stories/address-block.stories.mdx:3]: It doesn't have any designer-facing documentation. It's just an example of what it looks like (and the code example of how to make it happen).

File: packages/formation-react/stories/form-controls.stories.mdx (hunk 2 above)
Comment (cvalarida) [reply packages/formation-react/stories/form-controls.stories.mdx:3]: Oh, I did miss this one. Good call. :+1:

File: packages/formation-react/src/components/Breadcrumbs/usage.stories.mdx (hunk 3 above)
Comment (cvalarida) [reply packages/formation-react/src/components/Breadcrumbs/usage.stories.mdx:3]: I couldn't figure out the audience. It didn't seem useful for either engineers or designers, so I opted to remove it and see if anybody cared. :see_no_evil:

File: packages/formation-react/stories/form-controls.stories.mdx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { Canvas, Meta } from '@storybook/addon-docs/blocks';
 
-<Meta title="Library/Form controls/Usage" />
+<Meta title="Designer Documentation/Form controls/Usage" />
```
Comment (bkjohnson) [inline packages/formation-react/stories/form-controls.stories.mdx:3]: Do we still want the `Usage` in here?

**Review by bkjohnson — APPROVED** [review]
I was assuming that all `*.mdx` files would be `Designer Documentation`, but I guess that doesn't need to be the case.

File: packages/formation-react/stories/form-controls.stories.mdx (hunk 4 above)
Comment (cvalarida) [reply packages/formation-react/stories/form-controls.stories.mdx:3]: :man_facepalming: Nope!
